src/lib/hasher/hasher.ts

```diff
@@ -168,14 +174,82 @@ export function getLeafHasher<V extends CCIPVersion = CCIPVersion>({
   destChainSelector,
   onRamp,
   version,
-}: Lane<V>): LeafHasher {
+}: Lane<V>): LeafHasher<V> {
   switch (version) {
     case CCIPVersion.V1_2:
     case CCIPVersion.V1_5:
-      return getV12LeafHasher(sourceChainSelector, destChainSelector, onRamp)
+      return getV12LeafHasher(sourceChainSelector, destChainSelector, onRamp) as LeafHasher<V>
     case CCIPVersion.V1_6:
-      return getV16LeafHasher(sourceChainSelector, destChainSelector, onRamp)
+      return getV16LeafHasher(sourceChainSelector, destChainSelector, onRamp) as LeafHasher<V>
     default:
       throw new Error(`Unsupported CCIP version: ${version}`)
   }
 }
+
+export type AptosLeafHasher = (message: AptosCCIPMessage) => string
+
+export const makeAptosHasher =
```

Comment:
Nit: please, make naming compatible with the other leafhashers; you can assume you only need to implement it for V1_6+